src/modules/schema.js

```diff
@@ -0,0 +1,240 @@
+import { randomAsHex, encodeAddress, decodeAddress } from '@polkadot/util-crypto';
+import { u8aToU8a, u8aToString, u8aToHex } from '@polkadot/util';
+import { validate } from 'jsonschema';
+import axios from 'axios';
+
+import { getHexIdentifierFromDID } from '../utils/did';
+import { getSignatureFromKeyringPair } from '../utils/misc';
+import { isHexWithGivenByteSize } from '../utils/codec';
+import Signature from '../signatures/signature';
+
+import { DockBlobIdByteSize } from './blob';
+
+// Supported schemas
+import JSONSchema07 from '../utils/vc/schemas/schema-draft-07';
+
+// Blob qualifier
+export const BlobQualifier = 'blob:dock:';
+
+// Size of the blob id in bytes
+export const BLOB_ID_MAX_BYTE_SIZE = 32;
+
+// Maximum size of the blob in bytes
+// implementer may choose to implement this as a dynamic config option settable with the `parameter_type!` macro
+export const BLOB_MAX_BYTE_SIZE = 1024;
+
+/**
+ * Check if the given identifier is 32 byte hex
+ * @param {string} identifier - The identifier to check.
+ * @return {void} Throws exception if invalid identifier
+ */
+export function validateBlobDIDHexIdentifier(identifier) {
+  // Byte size of the Dock DID identifier, i.e. the `DockDIDQualifier` is not counted.
+  if (!isHexWithGivenByteSize(identifier, BLOB_ID_MAX_BYTE_SIZE)) {
+    throw new Error(`DID identifier must be ${BLOB_ID_MAX_BYTE_SIZE} bytes`);
+  }
+}
+
+/**
+ * Gets the hexadecimal value of the given DID.
+ * @param {string} did -  The DID can be passed as fully qualified DID like `blob:dock:<SS58 string>` or
+ * a 32 byte hex string
+ * @return {string} Returns the hexadecimal representation of the DID.
+ */
+export function getHexIdentifierFromBlobDID(did) {
+  if (did.startsWith(BlobQualifier)) {
+    // Fully qualified DID. Remove the qualifier
+    const ss58Did = did.slice(BlobQualifier.length);
+    try {
+      const hex = u8aToHex(decodeAddress(ss58Did));
+      // 2 characters for `0x` and 2*byte size of DID
+      if (hex.length !== (2 + 2 * BLOB_ID_MAX_BYTE_SIZE)) {
+        throw new Error('Unexpected byte size');
+      }
+      return hex;
+    } catch (e) {
+      throw new Error(`Invalid SS58 DID ${did}. ${e}`);
+    }
+  } else {
+    try {
+      // Check if hex and of correct size and return the hex value if successful.
+      validateBlobDIDHexIdentifier(did);
+      return did;
+    } catch (e) {
+      // Cannot parse as hex
+      throw new Error(`Invalid hexadecimal DID ${did}. ${e}`);
+    }
+  }
+}
+
+/**
+ * Generates a random schema ID
+ * @returns {string}
+ */
+export function createNewSchemaID() {
```

Comment:
Why not just call `createNewDockBlobId` inside?

Reply:
method was added/used before blob module was integrated, could do that